Flakey TestPprofServer

```
--- FAIL: TestPprofServer (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0xae0d95]

goroutine 5 [running]:
testing.tRunner.func1.2({0xc37de0, 0x1287690})
        /home/omer/GoLand-2022.3.2/sdk/go1.20.1/src/testing/testing.go:1526 +0x24e
testing.tRunner.func1()
        /home/omer/GoLand-2022.3.2/sdk/go1.20.1/src/testing/testing.go:1529 +0x39f
panic({0xc37de0, 0x1287690})
        /home/omer/GoLand-2022.3.2/sdk/go1.20.1/src/runtime/panic.go:884 +0x213
github.com/NethermindEth/juno/pprof_test.waitForServerReady(0xc000288680, {0xc000046db0, 0x22}, 0x1?)
        /home/omer/Documents/juno/pprof/pprof_test.go:50 +0xb5
github.com/NethermindEth/juno/pprof_test.TestPprofServer(0xc000288680)
        /home/omer/Documents/juno/pprof/pprof_test.go:28 +0x34e
testing.tRunner(0xc000288680, 0xd56a38)
        /home/omer/GoLand-2022.3.2/sdk/go1.20.1/src/testing/testing.go:1576 +0x10b
created by testing.(*T).Run
        /home/omer/GoLand-2022.3.2/sdk/go1.20.1/src/testing/testing.go:1629 +0x3ea
```

--- a/pprof/pprof_test.go
+++ b/pprof/pprof_test.go
@@ -43,11 +43,12 @@ func waitForServerReady(t *testing.T, url string, timeout time.Duration) {
 		require.NoError(t, reqErr)
 
 		resp, err := client.Do(req)
-		if err == nil && resp.StatusCode == http.StatusOK {
+		if resp != nil && resp.Body != nil {
 			resp.Body.Close()
+		}
+		if err == nil && resp.StatusCode == http.StatusOK {
 			return
 		}
-		resp.Body.Close()
 
 		require.Greaterf(t, timeout, start, "server is not ready after %v", timeout)
 

